### PR TITLE
Add INQURE annotations to JointBERT

### LINKED ISSUE #208: Add INQURE annotations to JointBERT

### PR DESCRIPTION
**Pull request by IKostric**
Fixes #208

### CODE REVIEW

File: moviebot/nlu/annotation/joint_bert/dataset.py [hunk 1]
```diff
@@ -153,20 +155,26 @@ def _tokenize_and_label(
...
+                labels.extend(
+                    [_IGNORE_INDEX] * self._num_inside_word_tokens(word)
                 )
-                labels.extend([_IGNORE_INDEX] * self._num_word_tokens(word) - 1)
             start_idx = index + len(slot_text)
 
         for word in text[start_idx:].split():
             labels.append(JointBERTSlot.to_index("OUT"))
-            labels.extend([_IGNORE_INDEX] * self._num_word_tokens(word) - 1)
+            labels.extend([_IGNORE_INDEX] * self._num_inside_word_tokens(word))
```
Comment (NoB0) [inline moviebot/nlu/annotation/joint_bert/dataset.py:177]: Just a clarification question, here you only keep the id of the first token for each word right?

File: moviebot/nlu/annotation/joint_bert/joint_bert_train.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 Batch = Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]
 
 _MODEL_OUTPUT_PATH = "models/joint_bert"
-_DATA_PATH = "data/training/nlu/utterances.yaml"
+_DATA_PATH = "data/training/utterances.yaml"
 
 _IGNORE_INDEX = -100
```
Comment (NoB0) [inline moviebot/nlu/annotation/joint_bert/joint_bert_train.py:27]: It looks like the same variable is defined in `dataset.py`, from my understanding these need to have the same value. So, I would suggest to keep only one and import it in the otehr file.

File: moviebot/nlu/neural_nlu.py [hunk 3]
```diff
@@ -61,22 +61,32 @@ def generate_dacts(
...
                 continue
+
+            if intent == UserIntents.INQUIRE and "INQUIRE" not in slot["slot"]:
+                continue
+            if (
+                intent in (UserIntents.REMOVE_PREFERENCE, UserIntents.INQUIRE)
+                and "PREFERENCE" not in slot["slot"]
+            ):
+                continue
+            slot_name = Slots[slot["slot"].split("_")[1]].value
```
Comment (NoB0) [inline moviebot/nlu/neural_nlu.py:80]: To be checked but should you take the last part of the split? For example if the slot is `B_PREFERENCE_GENRES` then with the current implementation `PREFERENCE` would be used to find the slot name. 


```suggestion
            slot_name = Slots[slot["slot"].split("_")[-1]].value
```

**Review by NoB0 — COMMENTED** [review]
Just a few questions to look at before approving.

File: moviebot/nlu/annotation/joint_bert/dataset.py (hunk 1 above)
Comment (IKostric) [reply moviebot/nlu/annotation/joint_bert/dataset.py:177]: Yes. I have extended the docstrings with an example.

File: moviebot/nlu/annotation/joint_bert/joint_bert_train.py (hunk 2 above)
Comment (IKostric) [reply moviebot/nlu/annotation/joint_bert/joint_bert_train.py:27]: Done.

File: moviebot/nlu/neural_nlu.py (hunk 3 above)
Comment (IKostric) [reply moviebot/nlu/neural_nlu.py:80]: Good catch.

**Review by NoB0 — APPROVED** [review]
LGTM.

**Pull request merged**